api/rest/client/lbclient.go

```diff
@@ -0,0 +1,383 @@
+package client
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	cid "github.com/ipfs/go-cid"
+	shell "github.com/ipfs/go-ipfs-api"
+	files "github.com/ipfs/go-ipfs-files"
+	"github.com/ipfs/ipfs-cluster/api"
+	peer "github.com/libp2p/go-libp2p-peer"
+)
+
+// Strategy values
+const (
+	Random Strategy = iota
+	RoundRobin
+)
+
+// Strategy represents load balancing strategy
+type Strategy int
+
+// loadBalancingClient a client to interact with IPFS Cluster APIs.
+// It balances the load by distributing it among peers.
+type loadBalancingClient struct {
+	clients  []Client
+	strategy Strategy
+	retries  int
+	length   int
+}
+
+// NewLBClient returens a new client that would load balance among
+// clients
+func NewLBClient(cfgs []*Config, strategy Strategy, retries int) (Client, error) {
+	var clients []Client
+	for _, cfg := range cfgs {
+		defaultClient, err := NewDefaultClient(cfg)
+		if err != nil {
+			return nil, err
+		}
+		clients = append(clients, defaultClient)
+	}
+
+	rand.Seed(time.Now().UnixNano())
+
+	return &loadBalancingClient{clients: clients, strategy: strategy, retries: retries, length: len(clients)}, nil
+}
+
+func (lc *loadBalancingClient) retry(call func(Client) error) error {
+	switch lc.strategy {
+	case Random:
+		return lc.retryRandom(0, call)
+	case RoundRobin:
+		return lc.retryRoundRobin(rand.Intn(len(lc.clients)), 0, call)
+	default:
+		return nil
+	}
+}
+
+func (lc *loadBalancingClient) retryRandom(count int, call func(Client) error) error {
+	err := call(lc.clients[rand.Intn(len(lc.clients))])
```

Comment:
The issue specified that retries should only happen in the face of a network error. The errors from clients are actually wrapping `api.Error` (or should be). A network error would be when:

* The error cannot be casted to `api.Error` (log this as this is abnormal), or
* The `error.Code == 0` (meaning no http request happened).